src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -726,13 +726,32 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(DHCPLease **lease, be32_t address, DHCPClientId *client_id, uint8_t *chaddr, be32_t gateway, usec_t expiration){
+        *lease = new0(DHCPLease, 1);
+        if (!lease)
```

Comment:
this oom check is flawed. should be `if (!*lease)`. But eitherway, you shouldn't manipulate the return parameter anyway until successful, see above.